tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -6,72 +6,76 @@ import (
-// It returns an AWS session
-func NewSession(credPath, credAccount, region *string) (*sessionaws.Session, error) {
-	if _, err := os.Stat(*credPath); os.IsNotExist(err) {
-		return nil, fmt.Errorf("failed to find AWS credentials from path '%v'", *credPath)
+// NewSession uses AWS credentials to create and returns a session for interacting with EC2.
+func NewSession(credentialPath, credentialAccountID, region string) (*awssession.Session, error) {
+	if _, err := os.Stat(credentialPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("failed to find AWS credentials from path '%v'", credentialPath)
 	}
-	sess := sessionaws.Must(sessionaws.NewSession(&aws.Config{
-		Credentials: credentials.NewSharedCredentials(*credPath, *credAccount),
-		Region:      aws.String(*region),
+	session := awssession.Must(awssession.NewSession(&aws.Config{
+		Credentials: credentials.NewSharedCredentials(credentialPath, credentialAccountID),
+		Region:      aws.String(region),
 	}))
-	return sess, nil
+	return session, nil
 }
 
-// CreateWindowsVM creates an Windows VM instance based on a given AWS session under the same VPC as the existing OpenShift cluster
-// It attaches existing OpenShift cluster worker security group and IAM
-// It also uses public subnet, attaches public ip, and a security group that allows traffic from VPC range and RDP access from my IP
-// It uses given image id, instance type, and key name
-// It creates a unique name tag for the instance using the same prefix as the OpenShift cluster name
-// It logs id and security group information of the created instance
-// At the end, it provides RDP access information in the commandline interface
-func (s *svc) CreateWindowsVM(imageId, instanceType, keyName *string) error {
+// CreateWindowsVM creates an Windows VM instance under the same VPC as the existing OpenShift cluster,
+// - attaches existing cloud-specific cluster worker security group and IAM to gain the same access as the linux
+// workers,
+// - uses public subnet,
+// - attaches public ip and a security group that allows traffic from VPC range and RDP access from user's IP,
+// - uses given image id, instance type, and key name,
+// - creates a unique name tag for the instance using the same prefix as the OpenShift cluster name, and
+// - logs id and security group information of the created instance in 'windows-node-installer.json' file at the
+// resourceTrackerDir.
+// The CreateWindowsVM outputs RDP access information in the commandline interface.
+func (s *cloudProviderInfo) CreateWindowsVM(imageId, instanceType, keyName string) error {
 	return nil
 }
 
-// DestroyWindowsVM destroys instances and security groups on AWS specified in windows-node-installer.json file
-// It consumes the file if succeeded
-func (s *svc) DestroyWindowsVM() error {
+// DestroyWindowsVM destroys instances and security groups on AWS specified in 'windows-node-installer.json' file.
```

Comment:
If you want to give the option to create and delete multiple instances, then the function names should reflect that.

Why would you ever delete multiple SGs? There should be only one SG per cluster.

Reply:
Are we supporting to create multiple VMs? We can easily do that, but I wonder if that is a feature we would want.

Reply:
Having a single worker is typically not done in production, so we eventually would need to support creating multiple VMs. . I am not suggesting we do that in this first cut but I want your comments to be consistent with what will be implemented.